### ISSUE TITLE
Localization

### ISSUE THREAD

**Issue by rnystrom** [rnystrom/GitHawk#567]
Are there any services that do string translation? Could we just Google translate it?

**Comment by Sherlouk** [comment]
So it's probably 100% overkill, but ages ago I used [CrowdIn](https://crowdin.com/page/ios-localization/) (free for open source) which has full support for iOS strings files etc

Provides a nice dashboard to see all the strings that need translating, interface for selecting all the different languages etc, as well as "machine recommended" (aka Google Translate) suggestions for easy translations

Found it very useful in past, but yeah maybe super overkill

(I should get sponsored by all these people I'm linking to 🤔)

**Comment by Sherlouk** [comment]
Oh and IIRC they have a Fastlane integration to make sure translations are always up-to-date 😉 

I know the way into your heart

**Comment by BasThomas** [comment]
I can translate to Dutch and probably find someone to translate to German 😇

**Comment by Palleas** [comment]
Happy to translate in French!

**Comment by rizwankce** [comment]
I can get help to translate to Chinese simplified. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by weyert** [comment]
I am happy to review to @BasThomas Dutch translation :) At work we are using `LocalizationKit` which allows to change resource strings without recompiling. It's written by a work colleague of mine.

**Comment by kimdv** [comment]
I'm in for danish translation 🚀

**Comment by BasThomas** [comment]
On longterm solutions: I was looking into making a script that would improve `genstrings`; instead of replacing, it would add new localizations and remove unused onces, while keeping the localizations already translated. Then I found [this](https://github.com/ndfred/xcode-tools/blob/master/update_strings.py), a Python script that tries to do exactly that.

I don't know what services like the ones that @Sherlouk mentioned would do over that, but would definitely like to know more about that. Otherwise, I'll do some testing with said script and see how it works.

**Comment by rnystrom** [comment]
Some recommended services after reaching out to Twitter:

- https://unbabel.com/
- https://www.transifex.com/
- https://poeditor.com/

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by weyert** [comment]
You also have https://lokalise.co which seems to be targeted towards iOS and Android apps

**Comment by Sherlouk** [comment]
@BasThomas so with my understanding of the tools, we'd actually want them both to work in harmony (although if I recall correctly CrowdIn and likely some of the ones Ryan linked would have it built-in).

The benefits of the web services is sort of the convenience factor? Essentially all the strings are (automatically) uploaded to their site, anybody (or certain people depending on the service) can then go in and add new translations, update translations, etc. So if new strings are added, it'll say "Dutch 90%, 5 new translations needed" then you can go in and provide the new translations.

Using a fastlane integration (or just command line) this will then pull down the new translated strings and put them into the bundle for releases.

So what does this really mean?
- Once a PR is merged into master, if new strings were added that weren't localized then through the website anybody can go and populate the new fields
- Anybody can comment on translations (nieuwe v nieuw) meaning reviews are much easier
- New languages can be added super easily, one string at a time by the community over time
- App Store metadata can be in the same place (#563)

All of the tools have minor differences, lokalise looks very good because you can upload screenshots for context. Most of them have "Clever translations" (where it tries to suggest translations).

For me it's making the process a bit simpler moving forward, I don't want to worry about adding new strings and then asking 10 people to provide translations in separate pull requests and having to manage all of that!

I may have missed a bunch of stuff so prompt if I'm not selling the idea enough 😂 (Had limited experience with it at work, was very helpful though!)

**Comment by BasThomas** [comment]
I'm convinced! Let's go with lokalise? Few questions:

- Do we need to do anything impactful for which we want to hold off Dutch translation now?
- If we go that route - and encourage others to contribute there - how do we "market" it?
- Regarding the above, we should then add useful comments to the `NSLocalizedString`s. I've actually been missing it during translation to Dutch, as sometimes I didn't have enough context to translate correctly - I'd suggest having an example here that would explain context.

**Comment by Sherlouk** [comment]
So nothing should stop us from getting the Dutch translations in, most of the tools have an import option which will take the strings files and just populate all the fields!

I'd say we open up the dashboard so anybody can hop on and translate, or at least review (you can make it so certain "trusted" people have the final say) - Link it in the README/Web/Twitter etc like we do for other things but also maybe setup a new label for "Localization Requests" and then put down the languages we want to priortise? Can obviously link to the right page and get tagging people and 🤞 

Definitely agree on comments!

Lokalise require messaging them for a free open-source account so it'd probably be up to Ryan to reach out and gain us access!

**Comment by weyert** [comment]
Great idea, I would happy to try to set it all up
